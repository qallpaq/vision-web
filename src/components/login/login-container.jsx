import React, {useEffect} from 'react'
import {loginUser} from "../../redux/thunk"
import {connect} from "react-redux"
import {Login} from "./login"
import {userIsUncreated} from "../../redux/actions"
import withAuthRedirect from "../withAuthRedirect"
import {compose} from "redux"


const LoginContainer = ({loginUser, userIsUncreated}) => {

    useEffect(() => {
        userIsUncreated()
    }, [])

    const Submit = (data) => {
        loginUser(data)
    }

    return <Login Submit={Submit}/>
}

export default compose(
    connect(null, {loginUser, userIsUncreated}),
    withAuthRedirect)
(LoginContainer)
