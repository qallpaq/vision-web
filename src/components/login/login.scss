@import "src/scss/constants";

.login__page {
  margin-top: $header-height + 100px;
}
