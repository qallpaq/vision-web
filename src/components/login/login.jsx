import React from 'react'
import LoginForm from "../login-form"
import './login.scss'


export const Login = ({Submit}) => {
    return (
        <div className='login__page'>
            <LoginForm onSubmit={Submit}/>
        </div>
    )
}
