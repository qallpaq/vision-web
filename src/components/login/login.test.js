import React from 'react'
import {Login} from "./login"
import {shallow} from "enzyme"


describe('Login', () => {
    const wrapper = shallow(<Login/>)

    it('should return snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should contain .login__page', () => {
        const result = wrapper.find('.login__page')
        expect(result).toHaveLength(1)
    })
})
