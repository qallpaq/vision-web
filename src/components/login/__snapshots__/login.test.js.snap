// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should return snapshot 1`] = `
<div
  className="login__page"
>
  <ReduxForm
    destroyOnUnmount={true}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="loginForm"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    submitAsSideEffect={false}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
  />
</div>
`;
